File: py3/640/540.py
from typing import List

class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        while l < r:
            m = l + (r - l) // 2
            m -= m & 1
            if nums[m] == nums[m + 1]:
                l = m + 2
            else:
                r = m
        return nums[r]

File: ts/640/540.ts
// 二分查找、位运算
function singleNonDuplicate(nums: number[]): number {
  let low = 0, high = nums.length - 1
  while (low < high) {
    const mid = ((high - low) >> 1) + low
    if (nums[mid] === nums[mid ^ 1]) {
      low = mid + 1
    } else {
      high = mid
    }
  }
  return nums[high]
};